macros/extract_descriptor.sql

```diff
@@ -1,4 +1,42 @@
 -- grab descriptor codes from namespaced descriptor values
 {% macro extract_descriptor(col) -%}
-    split_part({{ col }}, '#', -1)
+    
+    {%- set stripped_col = col.split(":")[-3] -%}
+    {%- set config = var('descriptors')[stripped_col] or None -%}
+    {%- set replace_with = config['replace_with'] or None -%}
+
+    -- if not configured to replace (default), split part from raw value
```

Comment:
All of the SQL comments are fully rendered when ran. We need to convert them to {# jinja comments. #}